checkpoint3-thealicat13

thealicat13 opened the pull request:
TODO: Consider making a parent class for teacher and apprentice teacher

Commit:
Add changed files

Commit:
Merge pull request #1 from thealicat13/checkpoint3-thealicat13

Add changed files

salogel42 commented on apprentice_teacher.rb:
```diff
@@ -1,4 +1,8 @@
+require_relative 'teacher'
+
+class ApprenticeTeacher < Teacher
+  undef_method :performance_rating
```
yeahhhh, this is pretty hacky.  Use a common superclass instead :)

salogel42 commented on apprentice_teacher.rb:
```diff
@@ -9,10 +13,6 @@ def initialize(options={})
-    "High five!"
-  end
-
   def set_phase(num)
```
Quite a bit of this is duplicating functionality that the superclass already has!  The whole point of the exercise was to avoid the repeated code.

salogel42 commented on teacher.rb:
```diff
@@ -7,10 +11,7 @@ def initialize(options={})
-
-  def offer_high_five
-    "High five!"
+    @performance_rating = 0
```
Why are you adding this functionality?

salogel42 commented on student.rb:
```diff
@@ -1,4 +1,7 @@
 class Student
+  include HighFivable
   attr_reader :age, :phase
   attr_accessor :name
```
What does the Student class have in common with the Teacher classes? How can we refactor it?

Commit:
Reset files

Commit:
Add the following:
- Educator class: parent class of Teacher and ApprenticeTeacher, move repetitive methods and attributes to parent, add lookup operator for subclass constants as attribute values
- Teacher, ApprenticeTeacher classes: make subclass of Educator, add constants
- Rspec: pass all rspec tests
NOTE: HighFivable module, 'undef_method' to raise error: leftover from original work done during checkpoint challenge
LEARNING OUTCOMES: recognize when a separate parent class should be made, solidify constant lookup using lookup operator, solidify requiring modules and class files for rspec tests or for class inheritance or module access, solidify how to exclude a child class from calling on a method/attribute in the parent class

Commit:
Merge pull request #2 from thealicat13/inher-practice

Inher practice

Commit:
Delete an unnecessary file

Commit:
Merge pull request #3 from thealicat13/cleanup

Delete an unnecessary file
